src/main/java/org/apache/commons/codec/net/QuotedPrintableCodec.java

```diff
@@ -208,6 +208,11 @@ public static final byte[] encodeQuotedPrintable(BitSet printable, final byte[]
         final int bytesLength = bytes.length;
 
         if (strict) {
+            // Ensure bytes have at least 3 bytes
+            if (bytesLength < 3) {
```

Comment:
Hello @arthurscchan 
If the number `3` is special, please refactor it into a constant and document it.